framework/src/org/checkerframework/framework/type/TypeFromExpressionVisitor.java

```diff
@@ -273,13 +273,17 @@ private void annotateArrayAsArray(AnnotatedArrayType result, NewArrayTree node,
     }
 
     /**
-     * The type of a NewClassTree is the type of the Identifier
-     * plus any explicit annotations (including polymorphic qualifiers)
-     * on the constructor.
+     * Creates an AnnotatedDeclaredType for the NewClassTree and adds one of:
```

Comment:
Is it `one of` also for type systems with multiple hierarchies? E.g. for Nullness, where Initialization and Nullness qualifiers exist at the same time.
What happens if the class is annotated `@A` and the constructor declaration `@B` and the instantiation `@C`, with A, B, and C from different hierarchies?